browser/ui/webui/brave_rewards_internals_ui.cc

```diff
@@ -137,6 +149,31 @@ void RewardsInternalsDOMHandler::HandleGetRewardsInternalsInfo(
                      weak_ptr_factory_.GetWeakPtr()));
 }
 
+void RewardsInternalsDOMHandler::HandleGetAdsInternalsInfo(
+    const base::ListValue* args) {
+  ads_service_->GetInternalsInfo(
+      base::BindOnce(&RewardsInternalsDOMHandler::OnGetAdsInternalsInfo,
+                     weak_ptr_factory_.GetWeakPtr()));
+}
+
+void RewardsInternalsDOMHandler::OnGetAdsInternalsInfo(
+    ads::InternalsInfoPtr info) {
+  if (!web_ui()->CanCallJavascript()) {
+    return;
+  }
+
+  base::DictionaryValue info_dict;
```

Comment:
This should be serialized from the ads lib using `base::JSONReader` as then we do not need to update Rewards internals for future changes